harness/determined/common/experimental/metrics.py

```diff
@@ -8,16 +8,16 @@
 
 @dataclasses.dataclass
 class TrialMetrics:
-    """
-    Specifies a metric that the trial reported.
+    """Specifies a metric that the trial reported.
 
     Attributes:
-        trial_id
-        trial_run_id
-        steps_completed
-        end_time
-        metrics
-        batch_metrics
+        trial_id: The ID of the trial that reported the metric.
+        trial_run_id: The ID of the trial run that reported the metric.
+        steps_completed: The number of steps that the trial had completed when the metric was
+            reported.
```

Comment:
We should probably specify that this is defined according to how the steps are defined in the trial (batches or epochs)

Reply:
I am not sure that's right. for core api metrics report, it's literally the `steps_completed` parameter passed to `report_metrics`. For Trial APIs, it'll be batches.

epochs value, if present, will be under `metrics["epochs"]`.

Reply:
`steps_completed` is an enigma to me. It's super confusing that we've written [a setter](https://github.com/determined-ai/determined/blob/8d2b5ce5b2134513a3c559d5545ad3e8c4fb3230/harness/determined/common/experimental/metrics.py#L51) for it.

Reply:
Oh I see, my mistake. And yeah in the case of coreapi usage, it could be whatever number you want IIRC

Reply:
ok -- I think I've taken this all into account with a new description.